File: library/src/main/java/org/fbme/lib/iec61499/declarations/FBTypeDeclaration.java
package org.fbme.lib.iec61499.declarations;

import org.fbme.lib.common.RootElement;
import org.fbme.lib.iec61499.descriptors.FBType;
import org.fbme.lib.iec61499.descriptors.FBTypeDescriptor;
import org.jetbrains.annotations.NotNull;

public interface FBTypeDeclaration extends FBInterfaceDeclarationWithAdapters, RootElement {

    default @NotNull FBTypeDescriptor getTypeDescriptor() {
        return new FBType(this);
    }
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/declarations/FBTypeDeclaration.kt
package org.fbme.lib.iec61499.declarations

import org.fbme.lib.common.RootElement
import org.fbme.lib.iec61499.descriptors.FBType
import org.fbme.lib.iec61499.descriptors.FBTypeDescriptor

sealed interface FBTypeDeclaration : FBInterfaceDeclarationWithAdapters, RootElement {
    val typeDescriptor: FBTypeDescriptor
        get() = FBType(this)
}
